packages/trader/src/Stores/Modules/Trading/trade-store.js

```diff
@@ -1039,6 +1044,23 @@ export default class TradeStore extends BaseStore {
             this.stop_out = limit_order?.stop_out?.order_amount;
         }
 
+        if (this.is_turbos && this.proposal_info) {
+            const contract_key = this.contract_type.toUpperCase();
+            if (this.proposal_info[contract_key]) {
+                const { barrier_choices, number_of_contracts } = this.proposal_info[contract_key];
+                this.turbos_barrier_choices = barrier_choices || [];
+                this.number_of_contracts = number_of_contracts ?? 0;
+                if (barrier_choices && !barrier_choices.includes(this.barrier_1)) {
+                    this.onChange({
+                        target: {
+                            name: 'barrier_1',
+                            value: barrier_choices[0],
+                        },
+                    });
+                }
```

Comment:
extracting necessary data for Turbos from proposal and setting them in trade-store.
Initial barrier taken from contracts_for gets outdated quickly so we have to check if it's applicable, i.e. present among barrier_choices. If it's not, we set the first value from barrier_choices as the barrier value.